### ISSUE TITLE
can animated encoding only be done with gifs?

### ISSUE THREAD

**Issue by databasedav** [image-rs/image#1909]
so a few of the formats (e.g. png, webp, gif) can be transformed into frames after decoding, but it looks like only the gif encoder has a convenient frames consumer method `.encode_frames`, but based on the avif encoder's use of `.write_all` in the `.write_image` method, does that mean i could somehow call `.write_image` with data from each of the frames to encode an animated avif?

**Comment by fintelia** [comment]
No, I think you're right that GIFs are the only animated format we currently support on the encoding side. The focus of this crate is on images rather than videos, but reach out if you'd be interested in working on a PR to add animated PNG/webp/etc.

**Comment by databasedav** [comment]
@fintelia yes i'm interested in working on such a pr, but i'll need some guidance :)

**Comment by fintelia** [comment]
Which format are you hoping to add? The first step would be to look at how the encoding for that format is currently implemented, and see what pieces are already in place for handling animation.

**Comment by databasedav** [comment]
@fintelia i think png would be a good start since i want it to run on wasm and can't find any pure rust encoders for webp or avif (yet), there's pure rust apng encoding support in https://github.com/poccariswet/apng, which can be used as inspiration

**Comment by fintelia** [comment]
In fact, given that the PNG crate already implements a [set_animated](https://docs.rs/png/latest/png/struct.Encoder.html#method.set_animated) method, I think a lot of the pieces may already be in place

**Comment by databasedav** [comment]
hi @fintelia just wanted to follow up on this, can u please provide some high level guidance on what needs to be done here? this is likely the lowest level thing i've ever done, but i think i can get it done with your help :)

**Comment by fintelia** [comment]
The first step would probably be writing a standalone app using the `png` crate to encode a APNG animation. I'd suggest looking through the docs.rs generated documentation and perhaps digging into the crate source code if the process 
seems unclear. Roughly, it looks like you need to be create an `Encoder`, call `set_animated` then `write_header`, use `write_image_data` to write each successive frame, and then call `finish`. Once you're able to generate valid animations, the next step would be porting it to the API used by this crate so that it can be merged here.

**Comment by stalkerg** [comment]
Any updates? Most interesting encode AVIF and WebP with animations, same as Python Pillow do it.

**Comment by fintelia** [comment]
We've now got a [pure-Rust WebP encoder](https://github.com/image-rs/image-webp) that can generate individual frames, so implementing animated WebP should be much more feasible now. But otherwise I think the state is pretty much the same.

**Comment by stalkerg** [comment]
unfortunately WebP is already in the past, and now the main focus to AVIF which can provide much better quality or size.
